From d04db46e05146d6096eac0f3f4a17a0bd9f1780c Mon Sep 17 00:00:00 2001
From: nyyu <mail@nyyu.dev>
Date: Mon, 29 Aug 2022 21:55:50 +0200
Subject: [PATCH] SurfaceFlinger: Don't cleanup resources from previous frame
 on virtual display

Causes bad lag on some legacy devices while WFD

(cherry picked from commit 356388ce2a98e0088cffb38903759bed2785d4e1)
---
 services/surfaceflinger/SurfaceFlinger.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 928dbb2a1b..d4613107f6 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -2910,7 +2910,9 @@ void SurfaceFlinger::processDisplayChanged(const wp<IBinder>& displayToken,
 
     // Recreate the DisplayDevice if the surface or sequence ID changed.
     if (currentBinder != drawingBinder || currentState.sequenceId != drawingState.sequenceId) {
+#ifndef DISABLE_POSTRENDER_CLEANUP
         getRenderEngine().cleanFramebufferCache();
+#endif
 
         if (const auto display = getDisplayDeviceLocked(displayToken)) {
             display->disconnect();
